astroquery/esa_hubble/core.py

```diff
@@ -0,0 +1,374 @@
+                   "),POSITION)=1  AND  PLANE.MAIN_SCIENCE_PLANE='true' "
+                   "ORDER BY PROPOSAL.PROPOSAL_ID "
+                   "DESC",
+                   # "PAGE": "1",
+                   # "PAGE_SIZE": "50",
+                   "RETURN_TYPE": str(output_format)}
+        result = urllib.parse.urlencode(payload,
+                                        quote_via=urllib.parse.quote_plus)
+        link = "".join((self.metadata_url, result))
+        if filename is None:
+            filename = "cone." + str(output_format)
+        print(link)
+        return self._handler.get_table(link, filename, output_format, verbose)
+
+    def query_target(self, name, filename=None, output_format='votable',
+                     verbose=False):
+        """ It executes a query over EHST and download the xml with the results
+
+            Parameters
+            ----------
+            name : string, target name to be requested, mandatory
+                Target name to be requested.
+            filename : string, file name to be used to store the metadata,
+            optional, default None
+                File name for the artifact
+            output_format : string, optional, default 'votable'
+                output format of the query
+            verbose : bool, optional, default 'False'
+                Flag to display information about the process
+
+            Returns
+            -------
+            Table with the result of the query. It downloads metadata
+            as a file.
+        """
+
+        initial = ("RESOURCE_CLASS=OBSERVATION&SELECTED_FIELDS=OBSERVATION"
+                   "&QUERY=(TARGET.TARGET_NAME=='")
+        final = "')&RETURN_TYPE=" + str(output_format)
+        link = self.metadata_url + initial + name + final
+        if filename is None:
+            filename = "target.xml"
+        print(link)
+        return self._handler.get_table(link, filename, output_format, verbose)
+
+    def query_hst_tap(self, query, output_file=None,
+                      output_format="votable", verbose=False):
+        """Launches a synchronous job to query the HST tap
+
+        Parameters
+        ----------
+        query : str, mandatory
+            query (adql) to be executed
+        output_file : str, optional, default None
+            file name where the results are saved if dumpToFile is True.
+            If this parameter is not provided, the jobid is used instead
+        output_format : str, optional, default 'votable'
+            results format
+        verbose : bool, optional, default 'False'
+            flag to display information about the process
+
+        Returns
+        -------
+        A Job object
+        """
+
+        return self._tap.launch_job(query=query, output_file=output_file,
+                                    output_format=output_format,
+                                    verbose=False,
+                                    dump_to_file=output_file is not None)
```

Comment:
Saving to file as an option is acceptable, but this query should return a `Table` object, not a `Job` object.

Reply:
Generally, though, we strongly prefer that users adopt a workflow like:
```
table = ESAHubble.query_hst_tap(parameters)
table.writeto('my_tablename.vot', format='ascii.votable')
```
over something like
```
table = ESAHubble.query_hst_tap(parameters, output_filename='my_votablename.vot')
```